custom_components/tuya_local/devices/multi_leaf_dy_107_thermostat.yaml

```diff
@@ -0,0 +1,196 @@
+name: Multi-leaf DY-107 thermostat

```

Comment:
Make this top level name generic (just "Thermostat" for example), as there may be other products that it matches.